### PR TITLE
[Upgrade Customer Scenario] Pytest conversion for Updated Satellite Sync API

### PR DESCRIPTION
**Pull request by devendra104**
The purpose of this PR to convert the Satellite Sync unit test(Customer upgrade scenario) to pytest.

Test Results:

**Pre-Upgrade**

```
$ pytest -m "pre_upgrade" tests/upgrades/test_satellitesync.py 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_satellitesync.py .                                                                                                                                                                      [100%]

================================================================================================ warnings summary =================================================================================================
================================================================================== 1 passed, 1 deselected, 7 warnings in 40.49s ===================================================================================

```
**Post-Upgrade**

```
$ pytest -m "post_upgrade" tests/upgrades/test_satellitesync.py 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_satellitesync.py .                                                                                                                                                                      [100%]

================================================================================================ warnings summary =================================================================================================
============================================================================= 1 passed, 1 deselected, 7 warnings in 97.88s (0:01:37) ==============================================================================

```

### CODE REVIEW

**Review by mshriver — APPROVED** [review]

File: tests/upgrades/test_satellitesync.py
Comment (pondrejk) [inline tests/upgrades/test_satellitesync.py:133]: Should we add some asserts to check if the delete actions were successful?

Comment (jyejare) [reply tests/upgrades/test_satellitesync.py:133]: This is not for assertions basically, its a tear-down.

But I am ok even if we assert there.

Comment (jyejare) [inline tests/upgrades/test_satellitesync.py:133]:
```suggestion
    exporting_cv_env_id = exporting_cv.read_json()['environments'][0]['id']
    importing_cv_env_id = importing_cv.read_json()['environments'][0]['id']
    exporting_cv.delete_from_environment(exporting_cv_env_id)
    importing_cv.delete_from_environment(importing_cv_env_id)
```

**Review by jyejare — CHANGES_REQUESTED** [review]
Comments to be addressed.

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: Fixed.

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py:91]: I will handle all nailgun kind of stuff in separate PR.

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: yes, I re-used one of the available fixtures.

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: Fixed.

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py:30]: I think we need to modify the design here no need to keep the class just for reading the scenarios. As we know the post-upgrade test case depends on pre-upgrade and it executes only after the pre-upgrade gets passed, so I think we can also skip uploading the pre-upgrade test results. We should need to target the post-upgrade test results only and each test case has proper docstring and I think that would be sufficient to understand the whole scenarios. 
What do you say?

Comment (jyejare) [reply tests/upgrades/test_satellitesync.py:30]: @devendra104 We should discuss this over a call. Lets meet.

File: tests/upgrades/test_satellitesync.py
Comment (jyejare) [reply tests/upgrades/test_satellitesync.py:99]: @devendra104 Not for organization. But a separate function scoped fixture that sets up the importing_org. Its just not org there, its product, repo etc. There should be one in UI tests for Foreman, check that.

File: tests/upgrades/test_satellitesync.py
Comment (jyejare) [reply tests/upgrades/test_satellitesync.py]: I didnt get what you have fixed there.

Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: I have added fixtures in pytest_fixtures/upgrade_fixtures.py, please refer to PR: https://github.com/SatelliteQE/robottelo/pull/8186

File: tests/upgrades/test_satellitesync.py
Comment (jyejare) [reply tests/upgrades/test_satellitesync.py:30]: @devendra104 As per discussion on the chat/Meet. I defiantly agree that we can rely on post-upgrade part completely to validate the scenario is pass/fail and of-course as you said above, the docstrings in postupgrade only can help us to upload the scenario as a Test and its results in Polarion.

But my only intention here is there should be something that holds the pre and post funcs as a `Upgrade Scenario` and that is achievable with putting pre and post test_functions in a one-liner class definition. This is not going to degrade or upgrade the performance but this will help to identify a one scenario easily. The code will also look more formatted and ease on eyes. 

Hope that helps you.

Comment (devendra104) [reply tests/upgrades/test_satellitesync.py:30]: Fixed.

**Comment by devendra104** [comment]
Test Results:

**Pre-Upgrade**
```
$ pytest -m "pre_upgrade" tests/upgrades/test_satellitesync.py::TestSatelliteSync
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_satellitesync.py .                                                                                                                                                                      [100%]

================================================================================================ warnings summary =================================================================================================
================================================================================== 1 passed, 1 deselected, 7 warnings in 53.03s ===================================================================================
```
**Post-Upgrade**

```
$ pytest -m "post_upgrade" tests/upgrades/test_satellitesync.py::TestSatelliteSync
=============================================================================================== test session starts ===============================================================================================
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_satellitesync.py .   
============================================================================= 1 passed, 1 deselected, 7 warnings in 95.62s (0:01:35) ==============================================================================

```

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: Created common fixtures that would be applicable for CLI test cases too.

File: tests/upgrades/test_satellitesync.py
Comment (jyejare) [reply tests/upgrades/test_satellitesync.py]: @devendra104 Do u have a plan to merge this PR once #8186 is merged since you would want to rebase and use that fixture in this PR ?

If thats the plan move this PR to draft or mark it as Do Not Merge.

File: tests/upgrades/test_satellitesync.py
Comment (jyejare) [reply tests/upgrades/test_satellitesync.py]: @devendra104 Anyways this is a non-blocking comment ! Feel free to work on it out of this PR.

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: fixed

File: tests/upgrades/test_satellitesync.py
Comment (devendra104) [reply tests/upgrades/test_satellitesync.py]: @jyejare I think we are good with both, Can you please review and merge #8186 first?

Comment (jyejare) [reply tests/upgrades/test_satellitesync.py]: Merged #8186 as all my comments were satisfied.

File: pytest_fixtures/api_fixtures.py
Comment (jyejare) [inline pytest_fixtures/api_fixtures.py:525]: Need not to explicitly mention its function scoped fixture.

Just:
```suggestion
@pytest.fixture
```
is enough.

**Review by jyejare — APPROVED** [review]

File: pytest_fixtures/api_fixtures.py
Comment (devendra104) [reply pytest_fixtures/api_fixtures.py]: Thanks @jyejare fixed.

**Pull request merged**
